docs/rfcs/0002-ecommerce.md

```diff
@@ -0,0 +1,176 @@
+information about multiple purchased items. However since the user can only purchase one course at
+once, there is only one `Line` per `Order` in MicroMasters.
+
+It's a requirement for mitxpro to support multiple items per order so we'll need a cart or basket
+for the user to keep track of their order. edX uses a `Basket` object for this purpose, which subclasses
+the `AbstractBasket` class from django-oscar. When a user clicks the button to verify a course, they
+are redirected to the `basket/add/?sku=SKU` URL which is handled by django-oscar to add the
+item to the basket.
+
+##### Recommendation
+
+Whether or not we want to use django-oscar, we will need models like these (using django-oscar names here):
+ - `Order` for an instance of a purchase
+ - `Basket` for a user's shopping cart. This should be connected to `Order`.
+ - `Line` for an item in an `Basket`
+ - `Product` to represent a purchasable item. In MicroMasters we skip this and 
+ instead set the course key in the `Line` directly but should have this object
+ to provide future flexibility and to let us decouple the price and the course information.
+ - `Voucher` to describe a coupon and how it can be used. A foreign key to `Basket` can be used
+ to indicate that a user wants to use the coupon for the purchase.
+ - `Receipt` to store CyberSource post-back information.
+ 
+And maybe something for purchase orders as well? We need some way to distinguish those from
+CyberSource purchases.
+
+Do we need audit tables? If so I think we can do this similar to how we did it for MicroMasters.
+
+#### django-oscar
+
+edX builds off of many of the abstract models that django-oscar provides, and they also subclass
+some of the django-oscar views. It makes sense to ask if we want to make use of these models, views or
+other functions rather than rolling our own like we did with MicroMasters.
+
+I think the big question is whether there is any functionality that could be useful and if using it saves times compared
+to rolling our own ecommerce infrastructure like we do on MicroMasters. While edX makes extensive use of django-oscar,
+they also have a lot of code which is written on top of django-oscar to handle their cases.
+
+##### Recommendation
+
+???
+
+#### Purchase Orders
+
+In addition to allowing end users to purchase courses, our solution will also
+need to handle purchase orders for bulk purchases. We should make sure to record this information
+in the same place that we record purchases via CyberSource so that it's simple to create reports from
+the information.
+
+We didn't add any support for purchase orders to MicroMasters. I think we manually added information
+via Django shell to the `Order` model. We will need an admin site to allow admins to fill in purchase
+order information so that they don't manipulate these models directly.
+
+##### Recommendation
+
+If we roll our own we should use the Django admin for this. django-oscar probably has an interface
```

Comment:
We might not need to worry about the logic for handling purchase orders for a while, but I think it will be helpful to understand who would be fulfilling these orders, how they have processed purchase orders in the past, and how they would prefer to process them in this app. Django admin might not end up being a good tool for creating a purchase order with a lot of line items

Reply:
That's true. I don't really have a sense for what Django admin can handle well, but we do need to be careful about the consistency of these models with each other so having a form that ensures this is a good idea.

Reply:
For MVP, we've agreed that I (or someone like me) will do the date entry for bulk orders. That buys us some time for figuring out whether django admin is adequate or if we need something more complex. 

To answer the question about fulfillment, we will be delivering enrollment code / 100%-off coupons -- for either course runs or programs.